Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/PolDiffILLReduction.py

```diff
@@ -0,0 +1,876 @@
+                                        dataE=dataE,
+                                        NSpec=mtd[conjoined_components].getItem(1).getNumberOfHistograms(),
+                                        OutputWorkspace='normalisation_ws')
+                    ws_name = '{0}_{1}'.format(tmp_name, entry_no)
+                    tmp_names.append(ws_name)
+
+                Divide(LHSWorkspace='normalisation_ws',
+                       RHSWorkspace=component_ws,
+                       OutputWorkspace=ws_name)
+
+            GroupWorkspaces(tmp_names, OutputWorkspace='det_efficiency')
+
+        single_efficiency_per_POL = False
+        if mtd[ws].getNumberOfEntries() != mtd['det_efficiency'].getNumberOfEntries():
+            single_efficiency_per_POL = True
+        for entry_no, entry in enumerate(mtd[ws]):
+            det_eff_entry_no = entry_no
+            if single_efficiency_per_POL:
+                det_eff_entry_no = det_eff_entry_no % nMeasurements
+            Multiply(LHSWorkspace=entry,
+                     RHSWorkspace=mtd['det_efficiency'].getItem(det_eff_entry_no),
+                     OutputWorkspace=entry)
+        return ws
+
+    def _sum_TOF_data(self, ws):
+        """Integrates intensities over all time channels or time-of-flight bins."""
+        tofUnits = self.getPropertyValue('TOFUnits')
+        if tofUnits == 'UncalibratedTime':
+            timeBinWidth = mtd[ws].getItem(0).getRun().getLogData('Detector.time_of_flight_0')
+            Multiply(LHSWorkspace=ws, RHSWorkspace=timeBinWidth, OutputWorkspace=ws)
+        if tofUnits == 'Energy':
+            energyBinWidth = 0 # placeholder
+            Multiply(LHSWorkspace=ws, RHSWorkspace=energyBinWidth, OutputWorkspace=ws)
+        Integration(InputWorkspace=ws, OutputWorkspace=ws)
+        return ws
+
+    def _output_vanadium(self, ws, n_atoms):
+        """Performs normalisation of the vanadium data to the expected cross-section."""
+        if self._mode == 'TOF':
+            ws = self._sum_TOF_data(ws)
+        Divide(LHSWorkspace=CreateSingleValuedWorkspace(0.404*n_atoms), RHSWorkspace=ws, OutputWorkspace=ws)
+        return ws
+
+    def _set_units(self, ws):
+        output_unit = self.getPropertyValue('OutputUnits')
+        if output_unit == 'TwoTheta':
+            if self.getProperty('SumScan').value and isinstance(mtd[ws], WorkspaceGroup) and mtd[ws].getNumberOfEntries() > 1:
+                self._merge_polarisations(ws)
+            else:
+                ConvertSpectrumAxis(InputWorkspace=ws, OutputWorkspace=ws, Target='SignedTheta')
+            ConvertAxisByFormula(InputWorkspace=ws, OutputWorkspace=ws, Axis='Y', Formula='-y')
+        elif output_unit == 'Q':
+            ConvertSpectrumAxis(InputWorkspace=ws, OutputWorkspace=ws, Target='ElasticQ',
+                                EFixed=self._experimentProperties['initial_energy'].value)
```

Comment:
Again, this could come from YIG calibrated lambda.

Reply:
It actually does come from YIG calibrated lambda and is calculated in `_read_experiment_properties` method.